### ISSUE TITLE
Refactor the use of BootImage for installation initrds

### ISSUE THREAD

**Issue by davidcassany** [OSInside/kiwi#873]
## Problem description

In OEM images if some installation media is required at least a couple of initrds are created. One the the OEM image itself and another one for the installation image. Those two initrds are created using the same instance of BootImage while the scope is not the same. The code related to that is a candidate for a refactor, it could be done in a simpler or clearer way. Currently this part is hard to follow and, in consequence, hard or potentially hard to maintain in the future.

Related with #858

**Comment by davidcassany** [comment]
I am starting to have look at this one. Unless I see a really straight forward implementation I will only document my thoughts and suggestions here.

**Comment by davidcassany** [comment]
@schaefi I am going to provide a PR to discuss it I really thing there is no longer the oemboot problem since we already dropped oemboot descriptions in OSInside/kiwi-descriptions@64ddd90d. So I am wondering if we should keep the door opened for any custom oemboot code out there since we are not maintaining it and it might silently break.

**Comment by schaefi** [comment]
sounds great, looking forward to it :+1:

**Comment by davidcassany** [comment]
Some trials on the test builds revealed my suspicions, with or without the patch custom boot on encrypted luks disk does not boot. Without luks oemboot code works fine. I am unsure what should we do from now on with this refactor, should we try to investigate whats going on for oemboot on luks? Or just assume the refactor is not breaking anything new?

On dracut all seams to work fine.

**Comment by schaefi** [comment]
Since we switched to dracut some years ago I expected that the custom oemboot code will be broken because the distro changes and nobody is testing or using oemboot anymore. Actually the only reference to a custom oemboot description lives in our integration tests and I'd like to keep the simple oem disk case to work as kind of a reference implementation if somebody wants to use this feature. However I expect this to be irrelevant anyway because we have dracut and that also supports custom extensions. So from my perspective it's more important to refactor the kiwi code such that the opportunity to build custom (non dracut based) initrds is still possible and works per integration test. If sophisticated layouts like LUKS needs special adaptions to a non dracut created initrd this should not be a priority anymore in my opinion.

As you found out LUKS+oemboot seems to be broken anyway and nobody is complaining I would leave it that way and offer help to customers when they stumble over it... which I think will never happen because we just use dracut.

> Or just assume the refactor is not breaking anything new?

if we can make that sure it would be ideal. I think what still has to work is the ```test-image-disk-legacy``` integration test.
It builds the most simple disk layout and install iso and we use this as a reference.

What do you think ?

**Comment by davidcassany** [comment]
All this makes sense to me. So the really important thing is that dracut base images are not affected by this change so special care on booting the integration tests is required for that. Regarding the oemboot and netboot I guess that booting our integration tests should be enough.

So to my eyes it looks like we should mostly move this code to Staging and boot all oem images.

> So from my perspective it's more important to refactor the kiwi code such that the opportunity to build custom (non dracut based) initrds is still possible and works per integration test.

yes agree
